src/gosling-brush/linear-brush-model.ts

```diff
@@ -0,0 +1,235 @@
+import type * as D3Selection from 'd3-selection';
+import type * as D3Drag from 'd3-drag';
+import { BrushAndMarkHighlightingStyle } from '@gosling.schema';
+
+const HIDDEN_BRUSH_EDGE_SIZE = 3;
+
+export type LinearBrushData = [LinearBrushBodyData, LinearBrushStartEdgeData, LinearBrushEndEdgeData];
+
+export type LinearBrushDataUnion = LinearBrushBodyData | LinearBrushStartEdgeData | LinearBrushEndEdgeData;
```

Comment:
Small nit. Rather than duplicating each of the types in `LinearBrushData` and `LinearBrushDataUnion`. It might be easier to just refer to the union _from_ the tuple. E.g.,

```typescript
import type { LinearBrushData } from './linear-brush-model';

function handler(data: LinearBrushData[number]) {
   // data: LinearBrushBodyData | LinearBrushStartEdgeData | LinearBrushEndEdgeData
}

Reply:
This is neat!